c/httpserver.c

```diff
@@ -2653,19 +2653,23 @@ static int safAuthenticate(HttpService *service, HttpRequest *request, AuthRespo
   } else if (authDataFound){
     ACEE *acee = NULL;
     strupcase(request->username); /* upfold username */
+    if (request->password == NULL) {
```

Comment:
Using the password field to indicate whether this is a passwordless auth seems like a bad idea. If for some unexpected reason that field is `NULL` (is it `NULL` if a user just doesn't provide a password in the basic auth?), you'll assume this is just a passwordless auth and will use `zisCheckUsername`, which might say everything is okay.

This seems like an error prone solution with implicit logic.

Regardless of whether this approach with client certificates is correct in the first place, using a separate variable to explicitly indicate what needs to be checked to authenticate would probably be better.